### ISSUE TITLE
Server crashes with 'Segmentation fault (core dumped)' when a client exit from SSL connection

### ISSUE THREAD

**Issue by bermond** [symless/synergy#5709]
### Operating Systems

Server: Arch Linux x86_64 Plasma Desktop
Client1: Arch Linux x86_64 LXQt Desktop
Client2: Linux Mint 17.3 i686 XFCE Edition
### Synergy Version

Server: 1.8.4
Client1: 1.8.4
Client2: 1.8.4 (manually compiled from git master at the time of 1.8.4 tagging)
### Steps to reproduce bug
1. Start `synergys` in server computer with SSL: `$ synergys --enable-crypto --no-daemon --debug DEBUG1 --log ~/.synergy/synergys.log`
2. Start `synergyc` in any (or both) client computer(s) with SSL: `synergyc --enable-crypto --no-daemon --debug DEBUG1 --log ~/.synergy/synergyc.log <server-address>`
3. Terminate any client. For example:
   `sudo systemctl reboot` or
   `killall -SIGTERM synergyc` or
   `killall -SIGKILL synergyc`
4. Server crashes with message: Segmentation fault (core dumped)
### Other info
- When did the problem start to occur? --> I'm not sure, but probably in 1.8.4.
- Is there a way to work around it? --> Yes, run without SSL (without `--enable-crypto`). This error occurs only when using SSL encryption.
- Does this bug prevent you from using Synergy entirely? --> No. It works, except for this server crash after a client exit. You need to restart the server everytime a client exits, what is really annoying when you run `synergys` as a background daemon.
- Probably it's not related, but I noticed that in 1.8.4 `synergy` (the GUI) doesn't allow using SSL (`Use SSL encryption` is grayed out in `Edit` -> `Preferences` -> `Network Security`). SSL works only at command line. I remember that this GUI option could be enabled in previous versions.
### Logs from Server and Client1

[synergys.log.txt](https://github.com/symless/synergy/files/552186/synergys.log.txt)
[synergyc.log.txt](https://github.com/symless/synergy/files/552187/synergyc.log.txt)

**Comment by faultylee** [comment]
I'm having the same issue, this is what I get from journalctl
```
Oct 27 11:16:23 xxx systemd-coredump[26369]: Process 25173 (synergys) of user 1000 dumped core.
                                                    
                                                    Stack trace of thread 25173:
                                                    #0  0x00000000000000b1 n/a (n/a)
```
From synergy `[2016-10-27T11:16:23] ERROR: process exited with error code: 11`
Server: 1.8.4 Linux (manually compiled from git master at the time of 1.8.4 tagging)
Client1: 1.8.4 Win10
Client2: 1.8.4  Win8.1

**Comment by Bazzek** [comment]
Having the same issue as well. I can confirm this started with 1.8.4, none of the 1.8.3 versions (beta and normal) had this problem.

Server: Xubuntu 16.10 with version 1.8.4
Client: Kubuntu 16.04 with version 1.8.4

**Comment by bermond** [comment]
This bug is fixed in Synergy 1.8.5.
Thanks to everyone who replied.
Closing.
